compiler/regex_test.go

```diff
@@ -45,8 +45,11 @@ func TestRegexCompiler(t *testing.T) {
 		{"urn:foo.bar.com:{.*{}", '{', '}', true, "", true},
 		{"urn:foo:<.*>", '<', '>', false, "urn:foo:bar:baz", false},
 
+		{`urn:foo:<user=(?!admin).*>`, '<', '>', false, "urn:foo:user=john", false},
+		{`urn:foo:<user=(?!admin).*>`, '<', '>', false, "urn:foo:user=admin", true},
+
```

Comment:
Could you please add a test for backwards compatible regex here? You could pick one of "named ascii character class [[:foo:]]" or "Python-style capture groups (P<name>re)" or both.

Reply:
OK, added.